src/common.rs

```diff
@@ -53,7 +53,7 @@ pub struct MidiInput {
     imp: MidiInputImpl
 }
 
-impl MidiInput {
+impl<'a> MidiInput {
```

Comment:
Why is the lifetime added to the `impl` instead of the function `connect` where it's needed?

Reply:
That was the first thing that came to my mind in order to fix the lifetime compile errors. For some reason I didn't think of making the lifetime `connect`-specific. Fixed in 8727d9163a8ba81f72a2049117f6889ac0df649f